### PR TITLE
feat: add script to check ownership request for the provided token address [AXL-2865]

### PR DESCRIPTION
**Pull request by blockchainguyy**

### CODE REVIEW

File: evm/check-ownership-request.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+async function checkAxelarDeployedToken(address, chain, provider) {
+    let isAxelarDeployed = false;
+    const apiUrl = `https://lcd-axelar.imperator.co/axelar/evm/v1beta1/token_info/${chain}?address=${address}`;
+
+    try {
+        const response = await axios.get(apiUrl);
+        const data = response.data;
+        isAxelarDeployed = data.confirmed === true && data.is_external === true;
```
Comment (ilyalukyanov) [inline evm/check-ownership-request.js:61]: shouldn't this be `data.is_external === false`?

File: evm/check-ownership-request.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+    program
+        .name('check-ownership-request')
+        .description('Before signing the message check that the contract address provided is not for AXL or any of our important contract');
+
+    addBaseOptions(program, { ignorePrivateKey: true, address: true });
+
+    program.addOption(new Option('-r, --rpc <rpc>', 'The rpc url for creating a provider to fetch gasOptions'));
+    program.addOption(
+        new Option('--addresses <addresses>', 'The Array of addresses for which the address to check against').env('ADDRESSES'),
```
Comment (ilyalukyanov) [inline evm/check-ownership-request.js:95]: What are these addresses going to be, for example? Where will they be taken from?

File: evm/check-ownership-request.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        printInfo(`Contract verification checks passed.`);
+    } catch (error) {
+        printError(`Error while checking contract address`, error.message);
+    }
+}
+
+async function checkContract(address, addressesArray, erc20) {
+    const name = await erc20.name();
+    const symbol = await erc20.symbol();
+    const regex = /axl/i;
```
Comment (ilyalukyanov) [inline evm/check-ownership-request.js:43]: Is this a reliable check? I can imagine how there could be both false-positives or false-negatives:
- If name or token symbol contain `axl` without any intention to mimic Axelar's naming. There are possible words where that can be a substring
- names/symbol of Axelar's tokens aren't guaranteed to contain `axl`.

File: evm/check-ownership-request.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    getDefaultProvider,
+    Contract,
+    constants: { AddressZero },
+} = ethers;
+
+const { addBaseOptions } = require('./cli-utils');
+const { printError, mainProcessor, printInfo } = require('./utils');
+
+const ERC20 = require('@axelar-network/axelar-cgp-solidity/artifacts/contracts/ERC20.sol/ERC20.json');
+const IInterchainToken = require('@axelar-network/interchain-token-service/interfaces/IInterchainToken.json');
```
Comment (deanamiel) [inline evm/check-ownership-request.js:16]: Use `getContractJSON` here

File: evm/check-ownership-request.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+async function processCommand(_, chain, options) {
+    const { rpc } = options;
+    let { address, addresses } = options;
```
Comment (deanamiel) [inline evm/check-ownership-request.js:20]: Add input validation checks for these params?

File: evm/check-ownership-request.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+async function main(options) {
+    await mainProcessor(options, processCommand);
+}
+
+if (require.main === module) {
+    const program = new Command();
+
+    program
+        .name('check-ownership-request')
+        .description('Before signing the message check that the contract address provided is not for AXL or any of our important contract');
```
Comment (deanamiel) [inline evm/check-ownership-request.js:89]:
```suggestion
        .description('Before signing the message, check that the contract address provided is not for AXL or any of our other important contracts');
```

File: evm/check-ownership-request.js (hunk 2 above)
Comment (deanamiel) [inline evm/check-ownership-request.js:95]:
```suggestion
        new Option('--addresses <addresses>', 'The Array of addresses that the address will be checked against').env('ADDRESSES'),
```

**Pull request closed without merging**